vowpalwabbit/array_parameters.h

```diff
@@ -107,6 +115,24 @@ class sparse_parameters
 
   bool not_null() { return (_weight_mask > 0 && !_map.empty()); }
 
+  void set_tag(uint64_t tag_hash) // function to store the tag hash for a feature and set it to true
+  {
+    _tag_info.tag_hash=tag_hash;
+    _tag_info.is_set=true;
+  }
+  
+  void turn_on_bit(uint64_t feature_index) // function to lookup a bit for a feature in the 32-bitset using the 5-bit tag hash and set it to 1.
```

Comment:
this should be:

```
 if(_tag_info.is_set) 
    {
      _feature_bit_vector[feature_index & _weight_mask][_tag_info.tag_hash]=1;
    }
```

to make sure we match the way the weights are accessed and stored (see `operator[]`) because that is the index we will need to lookup in our map later on when we do `is_activated`